### ISSUE TITLE
[Accessibility]: Compare page: Difficult to understand if diffs are additions/deletions

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35533]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34426

### Problem description

<img width="325" alt="image" src="https://user-images.githubusercontent.com/9516420/168785607-40651f10-2091-4789-b29d-0a5991d61ca9.png">

Navigating through this diff is difficult. A deletion is only marked with "hyphen", whereas a deletion is "plus".

### Expected behavior

We should add more descriptive labels that accurate descriptive the current line.

Possibly, we should use a `<th scope="row">` element to describe this row.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
